pkg/chartutil/values.go

```diff
@@ -263,18 +269,27 @@ func copyMap(src map[string]interface{}) map[string]interface{} {
 // coalesceValues builds up a values map for a particular chart.
 //
 // Values in v will override the values in the chart.
-func coalesceValues(c *chart.Chart, v map[string]interface{}) (map[string]interface{}, error) {
-	// If there are no values in the chart, we just return the given values
-	if c.Values == nil || c.Values.Raw == "" {
-		return v, nil
+func coalesceValues(c *chart.Chart, v map[string]interface{}, confType chart.ConfType) (map[string]interface{}, error) {
+	var nv Values
+	var err error
+	var vals *chart.Config
+
+	// Are we coalescing Secrets or Values?
+	if confType == chart.ConfType_SECRETS {
```

Comment:
The enum gets declared in `_proto/hapi/chart/config.proto`, with two type of VALUES or SECRETS and then the ConfType_SECRETS that you see is generated by protobuf and you can look at the object at `pkg/proto/hapi/chart/config.pb.go`